Pokemon Unity/Assets/Scripts/Save/GlobalSaveManager.cs

```diff
@@ -0,0 +1,275 @@
+﻿//Thanks to Omnicron for the reference to
+// https://pastebin.com/YY0XKMCQ
+//For BinarryFormatting any System.Serializable into a custom file
+
+using System.Runtime.Serialization.Formatters.Binary;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System;
+using UnityEngine.SceneManagement;
+using UnityEngine;
+
+public static class GlobalSaveManager
+{
+    private const string BuildVersion = "0.0.1";
+    private static string saveLocation = Environment.GetFolderPath(System.Environment.SpecialFolder.ApplicationData) + @"\Pokemon Unity\Saves\";
+
+    private static GameObject Player;
+    private static List<CustomSaveEvent> EventSaves = new List<CustomSaveEvent>();
+
+    private static EventListener EventListener = new EventListener();
+
+    public static void RegisterPlayer(GameObject player)
+    {
+        Player = player;
+        Debug.Log("Registered Player.");
+    }
+
+    /// <summary>
+    /// Registers an event into the EventSaves List<>
+    /// </summary>
+    /// <param name="customEvent">The event that needs to be registered</param>
+    public static void RegisterEvent(CustomSaveEvent customEvent)
+    {
+        EventSaves.Add(customEvent);
+        EventSaves = EventSaves.OrderBy(x => x.EventTime).ToList();
+        Debug.Log(customEvent.ToString());
+    }
+
+    /// <summary>
+    /// Get's the relevant CustomSaveEvents for the current Scene.
+    /// </summary>
+    /// <param name="sceneIndex">The scene index that the Player is currently on.</param>
+    /// <returns></returns>
+    public static List<CustomSaveEvent> GetRelaventSaveData(int sceneIndex)
+    {
+        return EventSaves.Where(x => x.SceneIndex == sceneIndex).ToList();
+    }
+
+    /// <summary>
+    /// Saves the game into the computer's %AppData% folder.
```

Comment:
Default should save back into directory of project/game. That way testing is easier than having to constantly access system folders. Also, if a user runs game, and they delete the game folder, they can delete residual residue of game.